web/src/components/vis/Heatmap.vue

```diff
@@ -454,6 +455,102 @@ export default {
       this.rnode = null;
       this.cnode = null;
     },
+
+    generateImage() {
```

Comment:
This code looks new (not moved from elsewhere). Does the same thing need to now be rendered in two different ways to support the interactive view and download? Will code need to change in two places to keep things looking the same?

Reply:
the heatmap is not just a simple SVG like other plots but
 * two SVGs for the dendograms
 * a HTML canvas  for the matrix
 * two divs with for the labels

Thus, cannot be handled in the same way as other plots